### PR TITLE
Answer:55 back button navigation

### PR DESCRIPTION
**Pull request by LMFinney**
Add an interface for components
Deactivate navigation based on the presence of a dialog (and sometimes with confirmation)

## Checklist for challenge submission

- [x] Start your PR message with Answer:${challenge_number}

## Warning:

- If you want feedback or review, you must support the project on GitHub:
  - [$5 per review](https://github.com/sponsors/tomalaforge)
  - [$25 for lifetime reviews](https://github.com/sponsors/tomalaforge)
  - Create one of more challenges
  - Have many valuable contributions to the project

Alternatively, you can still submit your PR to join the list of answered challenges or to be reviewed by a community member. 🔥

### CODE REVIEW

File: apps/angular/55-back-button-navigation/src/app/simple-action/simple-action.component.ts [hunk 1]
```diff
@@ -9,12 +10,26 @@ import { DialogComponent } from '../dialog/dialog.component';
...
+  }
 
   openDialog(): void {
-    this.#dialog.open(DialogComponent, {
+    this.#dialogRef = this.#dialog.open(DialogComponent, {
       width: '250px',
+      closeOnNavigation: false,
+    });
+
+    this.#dialogRef.afterClosed().subscribe(() => {
```
Comment (tomalaforge) [inline apps/angular/55-back-button-navigation/src/app/simple-action/simple-action.component.ts:31]: question: Do afterClosed trigger a complete event ?

**Comment by tomalaforge** [comment]
Hey @LMFinney, how are you ? 
I saw recently that you are looking a job ? Did you find one ? 
I hope you will get one, because you are a very good angular dev.

**Comment by LMFinney** [comment]
> Hey @LMFinney, how are you ? I saw recently that you are looking a job ? Did you find one ? I hope you will get one, because you are a very good angular dev.

Thanks for asking.
It's been only a week, but I've had one interview, and I have a few other leads. I think it'll be fine, but it was a rough week.

File: apps/angular/55-back-button-navigation/src/app/simple-action/simple-action.component.ts (hunk 1 above)
Comment (LMFinney) [reply apps/angular/55-back-button-navigation/src/app/simple-action/simple-action.component.ts:31]: Good point. I'm not sure.

To be safe, I just added an operator to clean it up.

**Comment by tomalaforge** [comment]
Yes I can imagine. That's the big downside of big american companies. Your job is not secure and you can be fired at any time. We get pay less in Europe but those things cannot happen. 
Good luck on your interviews, but I'm sure you'll find a job soon.

**Pull request closed without merging**